src/cart/crt_context.c

```diff
@@ -225,8 +226,9 @@ crt_context_provider_create(crt_context_t *crt_ctx, crt_provider_t provider, boo
 	D_RWLOCK_WRLOCK(&crt_gdata.cg_rwlock);
 	cur_ctx_num = crt_provider_get_cur_ctx_num(primary, provider);
 	max_ctx_num = crt_provider_get_max_ctx_num(primary, provider);
+	ctx_idx = crt_provider_get_ctx_idx(primary, provider);
 
-	if (cur_ctx_num >= max_ctx_num) {
+	if (ctx_idx < 0 || ctx_idx >= max_ctx_num || cur_ctx_num >= max_ctx_num) {
```

Comment:
is cur_ctx_num still needed here?

Reply:
We still have to increment as other place needs to know total number of contexts, so we already have cur_ctx_num; so might as well do this sanity check

Reply:
sure, but can't this be done as part of crt_provider_get_ctx_idx() instead of having to call 2 functions one to increment and one to get the available context ID?

Reply:
ah my bad sorry i got confused with crt_provider_inc_cur_ctx_num and crt_provider_get_cur_ctx_num.
anyway it just seems unnecessary here to call crt_provider_get_cur_ctx_num() on line 226 anymore.

Reply:
Agreed, i ll increment/decrement as part of get/put index